src/contracts/pods/EigenPod.sol

```diff
@@ -538,6 +491,78 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         return validatorInfo.restakedBalanceGwei * GWEI_TO_WEI;
     }
 
+    function _verifyBalanceUpdate(
+        uint64 oracleTimestamp,
+        uint40 validatorIndex,
+        bytes32 beaconStateRoot,
+        BeaconChainProofs.BalanceUpdateProof calldata balanceUpdateProof,
+        bytes32[] calldata validatorFields
+    ) internal {
+        
+        uint64 validatorBalance = balanceUpdateProof.balanceRoot.getBalanceAtIndex(validatorIndex);
+        bytes32 validatorPubkeyHash = validatorFields.getPubkeyHash();
+        ValidatorInfo memory validatorInfo = _validatorPubkeyHashToInfo[validatorPubkeyHash];
+
+        // 1. Balance updates should be more recent than the most recent update
```

Comment:
nit: the numbering of these checks is weird now, since (3) is now before (1)

Reply:
fixed